packages/editor/src/hooks/custom-sources-backwards-compatibility.js

```diff
@@ -106,13 +110,10 @@ const createWithMetaAttributeSource = ( metaAttributes ) =>
  */
 function shimAttributeSource( settings ) {
 	/** @type {WPMetaAttributeMapping} */
-	const metaAttributes = mapValues(
-		Object.fromEntries(
-			Object.entries( settings.attributes ?? {} ).filter(
-				( [ , { source } ] ) => source === 'meta'
-			)
-		),
-		'meta'
+	const metaAttributes = Object.fromEntries(
+		Object.entries( settings.attributes ?? {} )
+			.filter( ( [ , { source } ] ) => source === 'meta' )
+			.map( ( [ attributeKey, { meta } ] ) => [ attributeKey, meta ] )
```

Comment:
Already an array of entries, so it's straightforward enough to move the outer `mapValues()` to an inner, subsequent `.map()`.